### ISSUE TITLE
Error in `vsearch': corrupted double-linked list

### ISSUE THREAD

**Issue by frederic-mahe** [torognes/vsearch#70]
While trying to bisect another bug, I triggered this one:
```
head -n 30 biomarks_v4.fas > tmp.fas
vsearch -cluster_fast tmp.fas -id 0.97 -sizeout -msaout msa.fas -consout consensus.fas
vsearch v1.0.12_linux_x86_64, 15.6GB RAM, 4 cores
https://github.com/torognes/vsearch

Reading file tmp.fas 100%  
5716 nt in 15 seqs, min 374, max 387, avg 381
Indexing sequences 100% 
Masking 100%
Sorting by length 100%
Counting unique k-mers 100% 
Clustering 100%  
Writing clusters 100% 
Clusters: 14 Size min 1, max 2, avg 1.1
Singletons: 13, 86.7% of seqs, 92.9% of clusters
Multiple alignments 100% 
*** Error in `vsearch': corrupted double-linked list: 0x0000000000c914d0 ***
```

This is possibly linked to the issue 69 (gdb backtrace is the same):
```
(gdb) backtrace 
#0  0x00007ffff7029107 in __GI_raise (sig=sig@entry=6) at ../nptl/sysdeps/unix/sysv/linux/raise.c:56
#1  0x00007ffff702a4e8 in __GI_abort () at abort.c:89
#2  0x00007ffff7067204 in __libc_message (do_abort=do_abort@entry=1, fmt=fmt@entry=0x7ffff7159fe0 "*** Error in `%s': %s: 0x%s ***\n") at ../sysdeps/posix/libc_fatal.c:175
#3  0x00007ffff706c9de in malloc_printerr (action=1, str=0x7ffff7156064 "corrupted double-linked list", ptr=<optimized out>) at malloc.c:4996
#4  0x00007ffff706cc64 in malloc_consolidate (av=av@entry=0x7ffff7397620 <main_arena>) at malloc.c:4157
#5  0x00007ffff706d691 in _int_free (av=0x7ffff7397620 <main_arena>, p=<optimized out>, have_lock=0) at malloc.c:4057
#6  0x000000000040ee26 in cluster (dbname=<optimized out>, cmdline=<optimized out>, progheader=<optimized out>) at cluster.cc:1172
#7  0x000000000041bfb6 in main (argc=8, argv=0x7fffffffe848) at vsearch.cc:1711
```
Valgrind also incriminates the line 239 of msa.cc:
```
==20226== Invalid free() / delete / delete[] / realloc()
==20226==    at 0x4C29E90: free (vg_replace_malloc.c:473)
==20226==    by 0x40EE06: cluster(char*, char*, char*) (cluster.cc:1170)
==20226==    by 0x41BFB5: main (vsearch.cc:1711)
==20226==  Address 0x8da1720 is 0 bytes inside a block of size 5 free'd
==20226==    at 0x4C29E90: free (vg_replace_malloc.c:473)
==20226==    by 0x4126EE: msa(_IO_FILE*, _IO_FILE*, int, int, msa_target_s*) (msa.cc:239)
==20226==    by 0x40ED90: cluster(char*, char*, char*) (cluster.cc:1139)
==20226==    by 0x41BFB5: main (vsearch.cc:1711)
```

**Comment by torognes** [comment]
Is this with or without the patch from @xflouris?

**Comment by torognes** [comment]
I got the same error without @xflouris 's patch so I guess it was without it. It does indeed seem to be the same problem.

**Comment by torognes** [comment]
Fixed in VSEARCH version 1.0.13.
